// reducers/courses.js
import * as types from 'constants/types';

const initialState = {
    isFetching: false,
    courses: [],
    course: {},
    error: null,
};

export default (state = initialState, { type, payload }) => {

    switch (type) {
        case 'FETCH_COURSES_REQUESR':
            return { ...state, isFetching: true };
        case 'FETCH_COURSES_SUCCESS':
            return { ...state, isFetching: false, courses: payload };
        case 'FETCH_COURSE_SUCCESS':
            return { ...state, isFetching: false, course: payload };
        case 'FETCH_COURSES_FAILURE':
            return { ...state, isFetching: false, error: payload };
        default:
            return state;
    }
};

const initialStateCategories = {
    isFetching: false,
    categories: [],
    error: null,
};

export const coursesCategories = (state = initialStateCategories, { type, payload }) => {

    switch (type) {
        case types.courses.get.categories.request:
            return { ...state, isFetching: true };
        case types.courses.get.categories.success:
            return { ...state, isFetching: false, categories: payload };
        case types.courses.get.categories.failure:
            return { ...state, isFetching: false, error: payload };
        default:
            return state;
    }
};
