// constants/types.js
export const search = {
    request: 'search/request',
    success: 'search/success',
    failure: 'search/failure',
};

export const auth = {
    signIn: {
        request: 'auth/signIn/request',
        success: 'auth/signIn/success',
        failure: 'auth/signIn/failure',
    },
    signOut: {
        request: 'auth/signOut/request',
        success: 'auth/signOut/success',
        failure: 'auth/signOut/failure',
    }
};
export const registration = {
    request: 'registration/request',
    success: 'registration/success',
    failure: 'registration/failure',
};

export const posts = {
    get: {
        request: 'posts/fetch/get/request',
        success: 'posts/fetch/get/success',
        failure: 'posts/fetch/get/failure',
    },
    create: {
        request: 'posts/fetch/create/request',
        success: 'posts/fetch/create/success',
        failure: 'posts/fetch/create/failure',
    },
    like: 'post/like',
    unlike: 'post/unlike',
};

export const blog = {
    get: {
        articles: {
            request: 'blog/fetch/get/articles/request',
            success: 'blog/fetch/get/articles/success',
            failure: 'blog/fetch/get/articles/failure',
        },
        article: {
            request: 'blog/fetch/get/article/request',
            success: 'blog/fetch/get/article/success',
            failure: 'blog/fetch/get/article/failure',
        }
    },
};

export const gallery = {
    get: {
        images: {
            request: 'blog/fetch/get/images/request',
            success: 'blog/fetch/get/images/success',
            failure: 'blog/fetch/get/images/failure',
        },
    },
};

export const courses = {
    get: {
        request: 'courses/get/request',
        success: 'courses/get/success',
        failure: 'courses/get/failure',
        categories: {
            request: 'courses/get/categories/request',
            success: 'courses/get/categories/success',
            failure: 'courses/get/categories/failure',
        },
        course: {
            success: 'courses/get/course/success',
        }
    },
};

export const comments = {
    get: {
        request: 'comments/fetch/get/request',
        success: 'comments/fetch/get/success',
        failure: 'comments/fetch/get/failure',
    },
    create: {
        request: 'comments/fetch/create/request',
        success: 'comments/fetch/create/success',
        failure: 'comments/fetch/create/failure',
    },
    show: 'comments/show',
    toggle: 'comments/toggle',
};
