// constants/types.js
export const error = 'error';
export const loaded = 'loaded';
export const loading = 'loading';

export const auth = {
    login: {
        request: 'auth/login/request',
        success: 'auth/login/success',
        failure: 'auth/login/failure',
    },
    logout: {
        request: 'auth/logout/request',
        success: 'auth/logout/success',
        failure: 'auth/logout/failure',
    }
};

export const posts = {
    get: {
        request: 'posts/fetch/get/request',
        success: 'posts/fetch/get/success',
        failure: 'posts/fetch/get/failure',
    },
    create: {
        request: 'posts/fetch/create/request',
        success: 'posts/fetch/create/success',
        failure: 'posts/fetch/create/failure',
    },
    like: 'post/like',
    unlike: 'post/unlike',
};

export const blog = {
    get: {
        articles: {
            request: 'blog/fetch/get/articles/request',
            success: 'blog/fetch/get/articles/success',
            failure: 'blog/fetch/get/articles/failure',
        },
        article: {
            request: 'blog/fetch/get/article/request',
            success: 'blog/fetch/get/article/success',
            failure: 'blog/fetch/get/article/failure',
        }
    },
};

export const gallery = {
    get: {
        images: {
            request: 'blog/fetch/get/images/request',
            success: 'blog/fetch/get/images/success',
            failure: 'blog/fetch/get/images/failure',
        },
    },
};

export const comments = {
    get: {
        request: 'comments/fetch/get/request',
        success: 'comments/fetch/get/success',
        failure: 'comments/fetch/get/failure',
    },
    create: {
        request: 'comments/fetch/create/request',
        success: 'comments/fetch/create/success',
        failure: 'comments/fetch/create/failure',
    },
    show: 'comments/show',
    toggle: 'comments/toggle',
};
