// components/app/index.js
import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';

import Navigation from 'components/navigation';
import LandingPage from 'components/loading';
import SignUpPage from 'components/signup';
import SignInPage from 'components/signin';
import PasswordForgetPage from 'components/passwordForget';
import HomePage from 'components/home';
import AccountPage from 'components/account';
import AdminPage from 'components/admin';

import * as ROUTES from 'constants/routes';
import { withFirebase } from 'components/firebase';

class App extends Component {

    constructor(props) {
        super(props);

        this.state = {
            authUser: null,
        };
    }

    componentDidMount() {
        this.listener = this.props.firebase.auth.onAuthStateChanged(
            (authUser) => {
                authUser
                    ? this.setState({ authUser })
                    : this.setState({ authUser: null });
            }
        );
    }

    componentWillUnmount() {
        this.listener();
    }

    render() {
        return (
            <Router>
                <div>
                    <Navigation authUser={this.state.authUser} />

                    <hr />

                    <Route exact path={ROUTES.LANDING} component={LandingPage} />
                    <Route path={ROUTES.SIGN_UP} component={SignUpPage} />
                    <Route path={ROUTES.SIGN_IN} component={SignInPage} />
                    <Route path={ROUTES.PASSWORD_FORGET} component={PasswordForgetPage} />
                    <Route path={ROUTES.HOME} component={HomePage} />
                    <Route path={ROUTES.ACCOUNT} component={AccountPage} />
                    <Route path={ROUTES.ADMIN} component={AdminPage} />
                </div>
            </Router>
        );
    }
}

export default withFirebase(App);
