// components/Navbar/components/User/components/Account/index.js
import React, { Component } from 'react';

import { withFirebase } from 'components/firebase';
import SignOutLink from 'components/SignOutLink';

import './account.scss';

import avatar from './avatar.jpg';

class Account extends Component {

    constructor() {
        super();

        this.state = {
            displayMenu: false,
        };

        this.showMenu = this.showMenu.bind(this);
        this.hideMenu = this.hideMenu.bind(this);
    }

    hideMenu = () => {
        this.setState({ displayMenu: false }, () => {
            document.removeEventListener('click', this.hideMenu);
        });
    }

    showMenu = (event) => {
        this.setState({ displayMenu: true }, () => {
            document.addEventListener('click', this.hideMenu);
        });

        event.preventDefault();
    }

    render() {
        const { displayMenu } = this.state;

        return (
            <>
                <button
                    className="account"
                    onClick={this.showMenu}
                    type="button"
                >
                    <img src={avatar} alt="avatar" />
                </button>

                {displayMenu && (
                    <div className="account__menu">
                        <ul>
                            <li>
                                <a href="/profile">Profile</a>
                            </li>
                            <li>
                                <a href="/favorites">Favorites</a>
                            </li>
                            <li>
                                <SignOutLink />
                            </li>
                        </ul>
                    </div>
                )}
            </>
        );
    }
}

export default withFirebase(Account);
